.github/workflows/build-test-emscripten.yml

```diff
@@ -28,6 +28,20 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Collect runner's system stats
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: python3 scripts/devops/collect_runner_sys_stats.py
+
+      - name: Upload runner's system stats
+        uses: actions/upload-artifact@v4
+        with:
+          name: RunnerSysStats-emscripten-${{ matrix.config }}
+          path: RunnerSysStats-*
+          retention-days: 1
+        continue-on-error: true
```

Comment:
mb add new reusable worflow file, not to duplicate this steps in other workflows?